### PR TITLE
Add tests for process

### LINKED ISSUE #34: Support `std::process`
Support [`std::process`](https://doc.rust-lang.org/stable/std/process/index.html), including [`Command`](https://doc.rust-lang.org/stable/std/process/struct.Command.html).

 - Determine which system calls are needed. A first step might be to try to run tests/examples.rs under a mustang target, or [Rust's process tests](https://github.com/rust-lang/rust/blob/master/library/std/src/process/tests.rs), and see what unimplemented functions are called.
 - Add the system calls to [rsix](https://github.com/bytecodealliance/rsix).
 - Replace the `unimplemented!` code in c-scape with calls to `rsix`
 - For `fork`, we'll need to implement `__register_atfork` too, which will need something like a `static OnceCell<Mutex<Vec<...>>>` similar to `DTORS` in origin/src/programs.rs. It may make sense for this and `fork` itself to live in origin too, with c-scape providing just the C-ABI wrapper around it.
 - Add [Rusts's process tests](https://github.com/rust-lang/rust/blob/master/library/std/src/process/tests.rs) as tests/process.rs, with a comment at the top similar to tests/fs.rs.

If anyone's interested in working on any part of this, please leave a comment here! Or stop by [Zulip](https://bytecodealliance.zulipchat.com/#narrow/stream/217126-wasmtime). I'm happy to mentor, answer questions, explain what `fork` is, or, on the other hand, have an in-depth discussion about the relationship between `fork`, ambient authority, and operating system API design.

### PR DESCRIPTION
**Pull request by nivkner**
fixes #34, since this is the last step mentioned.

the cause of the failure in `smoke_failure` and `test_process_output_fail_to_start` is unrelated to mustang, wriiten https://github.com/rust-lang/rust/issues/90825 for that

it also seems that test commands that don't  include `--target=specs/x86_64-mustang-linux-gnu.json` are not actually testing mustang
(can be verified by modifying `c-scape` and then running the commands with the given enviroment variables).

it also seems that commands that contain `--target=specs/x86_64-mustang-linux-gnu.json` don't actually build on the desired architecture, since according to [the cargo reference](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget) the argument `--target` overrides `CARGO_BUILD_TARGET`.
(can be verified by adding a `compile_error` for arm and then running the commands with the given enviroment variables)

### CODE REVIEW

File: tests/process.rs [hunk 1]
```diff
@@ -0,0 +1,490 @@
...
+        Command::new("cmd").args(&["/C", "exit 0"]).spawn()
+    } else {
+        Command::new("true").spawn()
+    };
+    assert!(p.is_ok());
+    let mut p = p.unwrap();
+    assert!(p.wait().unwrap().success());
+}
+
+// NOTE(mustang): fails in the CI for a reason unreleated to mustang
```
Comment (sunfishcode) [inline tests/process.rs:30]: Could you add a reference to rust-lang/rust#90825 here, and below?

**Review by sunfishcode — COMMENTED** [review]
Very cool! Thanks for all your work working through the `fork` and `execve` issues.

**Comment by sunfishcode** [comment]
> it also seems that test commands that don't include `--target=specs/x86_64-mustang-linux-gnu.json` are not actually testing mustang (can be verified by modifying `c-scape` and then running the commands with the given enviroment variables).

Yes, that's right; I don't know of a way for tests to change their target.

The examples.rs test does utilize a mustang target to run the examples, which was useful before we had `Command` support, but now that that's in place, we can probably simplify things by just always running all the tests under a mustang target.

> it also seems that commands that contain `--target=specs/x86_64-mustang-linux-gnu.json` don't actually build on the desired architecture, since according to [the cargo reference](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget) the argument `--target` overrides `CARGO_BUILD_TARGET`. (can be verified by adding a `compile_error` for arm and then running the commands with the given enviroment variables)

I'm unclear on what you're saying here. Are you saying that compiling on arm or other hosts isn't working?

**Comment by nivkner** [comment]
> > it also seems that commands that contain `--target=specs/x86_64-mustang-linux-gnu.json` don't actually build on the desired architecture, since according to [the cargo reference](https://doc.rust-lang.org/cargo/reference/config.html#buildtarget) the argument `--target` overrides `CARGO_BUILD_TARGET`. (can be verified by adding a `compile_error` for arm and then running the commands with the given enviroment variables)
> 
> I'm unclear on what you're saying here. Are you saying that compiling on arm or other hosts isn't working?

that is what i am saying, the CI contains the `CARGO_BUILD_TARGET=${{ matrix.target }}`, where `matrix.target` is `armv5te-unknown-linux-gnueabi`, so without the `--target`, the `cargo test` command should build on arm (though without mustang), by adding `--target=specs/x86_64-mustang-linux-gnu.json`, the env var is overridden, and the command now build for exactly the specified target, i.e. `mustang` on x86-64.

i created https://github.com/nivkner/mustang/pull/3 to demonstrate that the tests are not being built for arm/aarch64.

maybe `RUST_TARGET_PATH` should be set in the CI and then `CARGO_BUILD_TARGET` could refer to `${{ matrix.mustang_target }}`,
and then the command would be `cargo +nightly test --verbose --release -Z build-std`

**Comment by nivkner** [comment]
opened https://github.com/sunfishcode/mustang/issues/66 to discuss the CI issue there instead, to remain on topic

**Pull request merged**